syncopy/specest/stft.py

```diff
@@ -0,0 +1,139 @@
+        lost on each side of the input signal. Defaults to `'zeros'`
+    detrend : str or `False`, optional
+        Optional detrending of the individual segments.
+        Sets `type` argument of :func: `~scipy.signal.detrend`,
+        acceptable are either `'constant'` or `'linear'`.
+        Defaults to  `False` such that no detrending is done.
+    padded : bool, optional
+         Specifies whether the input signal is zero-padded at the end to
+         make the signal fit exactly into an integer number of window
+         segments, so that all of the signal is included in the output.
+         Defaults to `True`. Padding occurs after boundary extension, if
+         `boundary` is not `None`, and `padded` is `True`, as is the
+         default.
+    axis : int, optional
+         Axis along which the STFT is computed; the default is over the
+         first axis (i.e. `axis=0`)
+
+    Returns
+    -------
+    ftr : :class:`numpy.ndarray`
+         Short-time fourier transform of the input `dat`
+         Per default the first axis corresponds to the segment times
+    freqs : :class:`numpy.ndarray`
+        Array of sampling frequencies
+    times : :class:`numpy.ndarray`
+        Array of segment times
+    """
+    # needed for stride tricks
+    # from here on axis=-1 is the data axis!
+    if dat.ndim > 1:
+        if axis != -1:
+            dat = np.moveaxis(dat, axis, -1)
+
+    # extend along time axis to fit in
+    # sliding windows at the edges
+    if boundary is not None:
+        zeros_shape = list(dat.shape)
+        zeros_shape[-1] = nperseg // 2
+        zeros = np.zeros(zeros_shape, dtype=dat.dtype)
+        dat = np.concatenate((zeros, dat, zeros), axis=-1)
+
+    # defaults to half window overlap
+    if noverlap is None:
+        noverlap = nperseg // 2
+    nstep = nperseg - noverlap
+
+    if padded:
+        # Pad to integer number of windowed segments
+        # I.e make x.shape[-1] = nperseg + (nseg-1)*nstep, with integer nseg
+        nadd = (-(dat.shape[-1]-nperseg) % nstep) % nperseg
+        zeros_shape = list(dat.shape[:-1]) + [nadd]
+        dat = np.concatenate((dat, np.zeros(zeros_shape)), axis=-1)
+
+    # Create strided array of data segments
+    if nperseg == 1 and noverlap == 0:
+        dat = dat[..., np.newaxis]
+    else:
+        # https://stackoverflow.com/a/5568169
+        step = nperseg - noverlap
+        shape = dat.shape[:-1] + ((dat.shape[-1] - noverlap) // step, nperseg)
+        strides = dat.strides[:-1] + (step * dat.strides[-1], dat.strides[-1])
+        dat = np.lib.stride_tricks.as_strided(dat, shape=shape,
+                                              strides=strides)
+    # dat now has shape (nChannels, nSamples, nperseg)
+
+    # detrend each segment separately
+    if detrend:
+        dat = sci_sig.detrend(dat, type=detrend, overwrite_data=True)
+
+    if window is not None:
+        # Apply window by multiplication
+        dat = window * dat
```

Comment:
Again I suggest using
```
dat *= window
```

Reply:
With this we are more strict than for example SciPy, which does it this way ;)